addon/initializers/component-styles.js

```diff
@@ -0,0 +1,36 @@
+import Ember from 'ember';
+import podNames from 'ember-component-css/pod-names';
+
+const {
+  Component,
+  ComponentLookup,
+} = Ember;
+
+export function initialize() {
+
+  ComponentLookup.reopen({
+    componentFor(name, owner) {
+      if (podNames[name] && !owner.application.hasRegistration('component:' + name)) {
```

Comment:
I had tried this it wasn't working for me. I think cause _lookupFactory does more then just return the constructor. I would have to call super again at the end, which calling it twice felt weird.